feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -298,149 +249,137 @@ func TestSwitchChip(t *testing.T) {
-		bpCapacity := component.IntegratedCircuit().BackplaneFacingCapacity()
-
-		totalCapacity := gnmi.Get(t, dut, bpCapacity.TotalOperationalCapacity().State())
-		t.Logf("Hardware card %s totalCapacity: %d", card, totalCapacity)
-		if totalCapacity <= 0 {
-			t.Errorf("bpCapacity.TotalOperationalCapacity().Get(t) for %q): got %v, want > 0", card, totalCapacity)
-		}
+			totalCapacity := gnmi.Get(t, dut, bpCapacity.TotalOperationalCapacity().State())
+			t.Logf("Hardware card %s totalCapacity: %d", card, totalCapacity)
+			if totalCapacity <= 0 {
+				t.Errorf("bpCapacity.TotalOperationalCapacity().Get(t) for %q): got %v, want > 0", card, totalCapacity)
+			}
 
-		total := gnmi.Get(t, dut, bpCapacity.Total().State())
-		t.Logf("Hardware card %s total: %d", card, totalCapacity)
-		if total <= 0 {
-			t.Errorf("bpCapacity.Total().Get(t) for %q): got %v, want > 0", card, total)
-		}
+			total := gnmi.Get(t, dut, bpCapacity.Total().State())
+			t.Logf("Hardware card %s total: %d", card, totalCapacity)
+			if total <= 0 {
+				t.Errorf("bpCapacity.Total().Get(t) for %q): got %v, want > 0", card, total)
+			}
 
-		if !gnmi.Lookup(t, dut, bpCapacity.AvailablePct().State()).IsPresent() {
-			t.Errorf("bpCapacity.AvailablePct() for %q): got none, want >= 0", card)
-		}
-		availablePct := gnmi.Get(t, dut, bpCapacity.AvailablePct().State())
-		t.Logf("Hardware card %s availablePct: %d", card, availablePct)
+			if !gnmi.Lookup(t, dut, bpCapacity.AvailablePct().State()).IsPresent() {
+				t.Errorf("bpCapacity.AvailablePct() for %q): got none, want >= 0", card)
+			}
+			availablePct := gnmi.Get(t, dut, bpCapacity.AvailablePct().State())
+			t.Logf("Hardware card %s availablePct: %d", card, availablePct)
 
-		if !gnmi.Lookup(t, dut, bpCapacity.ConsumedCapacity().State()).IsPresent() {
-			t.Errorf("bpCapacity.ConsumedCapacity() for %q): got none, want >= 0", card)
+			if !gnmi.Lookup(t, dut, bpCapacity.ConsumedCapacity().State()).IsPresent() {
+				t.Errorf("bpCapacity.ConsumedCapacity() for %q): got none, want >= 0", card)
+			}
+			consumedCapacity := gnmi.Get(t, dut, bpCapacity.ConsumedCapacity().State())
+			t.Logf("Hardware card %s consumedCapacity: %d", card, consumedCapacity)
 		}
-		consumedCapacity := gnmi.Get(t, dut, bpCapacity.ConsumedCapacity().State())
-		t.Logf("Hardware card %s consumedCapacity: %d", card, consumedCapacity)
 	}
 }
 
 func TestTempSensor(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 
-	r := regexp.MustCompile("^TempSensor")
-	sensors := findMatchedComponents(t, dut, r)
-	t.Logf("Found TempSensor list: %v", sensors)
-
+	sensors := components.FindComponentsByType(t, dut, oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_SENSOR)
+	t.Logf("Found Sensor list: %v", sensors)
 	if len(sensors) == 0 {
 		t.Fatalf("Get TempSensor list for %q: got 0, want > 0", dut.Model())
 	}
 
 	for _, sensor := range sensors {
-		t.Logf("Validate card %s", sensor)
-		component := gnmi.OC().Component(sensor)
+		r := regexp.MustCompile("Temp|TEMP")
```

Comment:
this is vendor dependent check and i have already addressed in a way all vendors can use please check https://github.com/openconfig/featureprofiles/pull/1277

Reply:
I would rather prefer this to be a vendor specific switch case inside the script rather than a runtime argument which may open up possibilities for test failures due to user error.